phoenix-core/src/main/java/org/apache/phoenix/compile/DeclareCursorCompiler.java

```diff
@@ -48,7 +48,13 @@
     public DeclareCursorCompiler(PhoenixStatement statement, Operation operation, QueryPlan queryPlan) {
         this.statement = statement;
         this.operation = operation;
-        this.queryPlan = queryPlan;
+        QueryPlan compilePlan = queryPlan;
+        try {
+            compilePlan = statement.getConnection().getQueryServices().getOptimizer().optimize(statement, queryPlan);
+        } catch (SQLException e) {
```

Comment:
I wanted to handle the exception to minimise the effect of this change in case the optimize() throws an Exception.
Even if we handle the exception here (in the DeclareCursorCompiler) later optimize() will be called on the CursorFetchPlan and the Exception would be thrown there so it doesn't really makes sense to handle it here probably.